Arcade Classics/Flixius/source/PlayState.hx

```diff
@@ -0,0 +1,644 @@
+		
+		var gfxMap:FlxSprite = new FlxSprite();
+		var arrMap:Array<Array<Int>> = [];
+		var arrDecoTop:Array<Array<Int>> = [];
+		var arrDecoBottom:Array<Array<Int>> = [];
+		_map = new FlxTilemap();
+		_mapDecoTop = new FlxTilemap();
+		_mapDecoBottom = new FlxTilemap();
+		
+		gfxMap.loadGraphic(AssetPaths.map__png);
+		
+		var e:Enemy;
+		var s:EnemySpinner;
+		var p:EnemyPod;
+		
+		for (y in 0...Std.int(gfxMap.height))
+		{
+			arrMap.push([]);
+			arrDecoTop.push([]);
+			arrDecoBottom.push([]);
+			for (x in 0...Std.int(gfxMap.width))
+			{
+				
+				if (gfxMap.pixels.getPixel(x,y) == 0x000000)
+				{
+					arrMap[y].push(rnd.int(1, 4));
+					arrDecoTop[y].push(0);
+					arrDecoBottom[y].push(0);
+				}
+				else if (gfxMap.pixels.getPixel(x, y) == 0x00ff00)
+				{
+					arrDecoBottom[y].push(rnd.int(1, 9));
+					arrDecoTop[y].push(0);
+					arrMap[y].push(0);
+				}
+				else if (gfxMap.pixels.getPixel(x, y) == 0xffff00)
+				{
+					arrDecoTop[y].push(rnd.int(1, 9)+9);
+					arrDecoBottom[y].push(0);
+					arrMap[y].push(0);
+				}
+				else if (gfxMap.pixels.getPixel(x, y) == 0xff0000)
+				{
+					e = new Enemy(x * 16, y * 16, this);
+					
+					_grpEnemies.add(e);
+					_grpEThrust.add(new Jet(e, 1));
+					arrMap[y].push(0);
+					arrDecoTop[y].push(0);
+					arrDecoBottom[y].push(0);
+				}
+				else if (gfxMap.pixels.getPixel(x, y) == 0xff00ff)
+				{
+					s = new EnemySpinner(x * 16, y * 16, this);
+					
+					_grpEnemiesSpinners.add(s);
+					
+					arrMap[y].push(0);
+					arrDecoTop[y].push(0);
+					arrDecoBottom[y].push(0);
+				}
+				else if (gfxMap.pixels.getPixel(x, y) == 0x00ffff)
+				{
+					p = new EnemyPod(x * 16, y * 16, this);
+					
+					_grpEnemiesPods.add(p);
+					
+					arrMap[y].push(0);
+					arrDecoTop[y].push(0);
+					arrDecoBottom[y].push(0);
+				}
+				else
+				{
+					arrMap[y].push(0);
+					arrDecoTop[y].push(0);
+					arrDecoBottom[y].push(0);
+				}
+			}
+		}
+		
+		_map.loadMapFrom2DArray(arrMap, AssetPaths.tiles__png, 16, 16, FlxTilemapAutoTiling.OFF, 1, 1, 1);
+		_mapDecoTop.loadMapFrom2DArray(arrDecoTop, AssetPaths.deco__png, 16, 16, FlxTilemapAutoTiling.OFF, 1, 1, 1);
+		_mapDecoBottom.loadMapFrom2DArray(arrDecoBottom, AssetPaths.deco__png, 16, 16, FlxTilemapAutoTiling.OFF, 1, 1, 1);
+		_mapDecoTop.y -= 4;
+		_mapDecoBottom.y += 4;
+		
+		
+		add(_map);
+		add(_mapDecoTop);
+		add(_mapDecoBottom);
+	}
+	
+	/**
+	 * Function that is called when this state is destroyed - you might want to 
+	 * consider setting all objects this state uses to null to help garbage collection.
+	 */
+	override public function destroy():Void
+	{
+		super.destroy();
+	}
+	
+	
+	public function returnFromSubState():Void
+	{
+		_showingReady = true;
+		#if flash
+		FlxG.sound.playMusic(AssetPaths.music__mp3);
+		#else
+		FlxG.sound.playMusic(AssetPaths.music__ogg);
+		#end
+		/* okay, I admit, this is ridiculous, but it works ;) I would probably find a nicer solution for a 'real' game lol */
```

Comment:
I don't feel comfortable having this abomination in a demo. ;) Why is this necessary?